### PR TITLE
Dark Mode

### PR DESCRIPTION
**Pull request by shinyichen**
* Added a dark mode switch on the right lower corner
* Remembers the setting in the local storage
* Uses system level dark mode preference if dark mode app setting not detected
* Dark mode CSS is applied based on `data-theme` attribute in the `body` tag.
* Dark mode CSS does not affect original CSS
* Corresponding updates for radio and checkboxes https://github.com/adsabs/ui-component-library/pull/16

Screen Shot 2021-05-18 at 9 33 27 AM
Screen Shot 2021-05-18 at 9 34 02 AM

### CODE REVIEW

File: src/js/dark-mode-switch.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    document.body.removeAttribute('data-theme');
+    darkSwitch.classList.remove('darkModeOn');
+    darkSwitch.setAttribute('title', 'Turn on dark mode');
+    if (save) localStorage.setItem('darkSwitch', 'off');
+  };
+
+  const init = () => {
+    // 1. check app setting
+    if (
+      localStorage.getItem('darkSwitch') !== null &&
```
Comment (thostetler) [inline src/js/dark-mode-switch.js:21]: We also technically already have a little wrapper around localstorage, you could bring it in in the header `define(['storage'], function (storage) { ... })`

**Review by thostetler — CHANGES_REQUESTED** [review]
Nice work on the styling, this all looks great! :)
I think we'll just want to move your script into an actual module so it gets loaded properly and not ahead of more important things, etc.
Let me know if you need any help getting that to work.

File: src/index.html [hunk 2]
```diff
@@ -107,6 +107,8 @@ <h3 id="basic-html-message">
         </article>
       </div>
     </div>
+    <div id="darkSwitch" class="darkmode-toggle" title="Turn on dark mode">🌓</div>
+    <script src="js/dark-mode-switch.js"></script>
```
[the remark this answers is not in the document]
Comment (shinyichen) [reply src/index.html]: Done

File: src/js/dark-mode-switch.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+(function() {
```
[the remark this answers is not in the document]
Comment (shinyichen) [reply src/js/dark-mode-switch.js]: Done

**Review by thostetler — APPROVED** [review]
Updated changes look good to me, I'll merge

**Pull request merged**